epub33/rs/index.html

```diff
@@ -1852,33 +1867,73 @@ <h4>Scripting Considerations</h4>
-
+				<p>
+					In handling untrusted scripts this specification recommends that Reading Systems establish 
+					a unique <a href="https://url.spec.whatwg.org/#origin">origin</a> [[URL]] allocated to each 
+					<a>EPUB Publication</a> (see <a href="#sec-scripted-content"></a>). Adopting this
+					approach isolates publications from each other, thereby limiting access to cookies, DOM storage, etc. Examples of Web APIs that are tied to the concept of “origin” include Web Storage [[?WEBSTORAGE]] and 
+					IndexedDB [[?INDEXEDDB]], which EPUB content documents can interact with via scripting.
+					This specification recommends the following practices:
+				</p>
 				<ul>
+
+					<li>
+						<p>
+							Although domain isolation is the most effective way to secure EPUB Publications in
+							Web-based Reading Systems, it is not always a realistic option. Web-based Reading
+							Systems must still maintain isolation in such cases, which may require limiting access
+							to scripting APIs that allow client-side data storage and access.
+						</p>
+					</li>
+					<li>
+						<p>
+							Reading Systems that allow users to add/remove publications from a managed library 
+							(a.k.a “bookshelf”) may maintain the publication’s unique origin when the publication is
+							removed and subsequently re-imported into the content library. Conversely, Reading
+							Systems may create a new unique origin for every newly-added publication.
+						</p>
+					</li>
+
+					<li>
+						<p>
+							The unique origin for a given publication (and by extension, for all its content documents)
+							must be preserved across "renders", i.e. when closing/re-opening publications, or when navigating back-and-forth between documents within an opened publication.
+						</p>
+					</li>
+
 					<li>
-						<p>Reading Systems must behave as if a unique <a href="https://url.spec.whatwg.org/#origin"
-								>origin</a> [[URL]] were allocated to each <a>EPUB Publication</a>. Adopting this
-							approach will isolate publications from each other, thereby limiting access to cookies, DOM
-							storage, etc.</p>
-						<div class="note">
-							<p>Although domain isolation is the most effective way to secure EPUB Publications in
-								Web-based Reading Systems, it is not always a realistic option. Web-based Reading
-								Systems must still maintain isolation in such cases, which may require limiting access
-								to scripting APIs that allow client-side data storage and access.</p>
-						</div>
-						<p>Reading Systems that enable scripting and network access should also include methods to
-							notify the user that network activity is occurring and/or that allow them to disable it.</p>
+						<p>
+							Reading Systems must ensure that the data tied to specific <a href="#confreq-rs-scripted-origin">origins</a>
+							(e.g., Web Storage [[?WEBSTORAGE]] or IndexedDB [[?INDEXEDDB]]) is preserved between "reading".
```

Comment:
Not sure what 'preserved between "reading"' means. Between reading sessions? Though also same comment as above - should we really be recommending this?

Reply:
I think "reading session" is a much better term. I will change this.